builder/relinker/relinker.py

```
@@ -0,0 +1,605 @@
+        
+        # Handle EXCLUDE_FILE logic using desc_to_lib mapping
+        for desc, lib in self.desc_to_lib.items():
+            index = '*%s:(EXCLUDE_FILE'%(lib)
+            if index in l and desc not in l:
+                # Collect all descriptors for this library
+                processed = set()
+                for m_desc, m_lib in self.desc_to_lib.items():
+                    m_index = '*%s:(EXCLUDE_FILE'%(m_lib)
+                    if m_index in l and m_desc not in l and m_desc not in processed:
+                        processed.add(m_desc)
+                        l = l.replace('EXCLUDE_FILE(', 'EXCLUDE_FILE(%s '%(m_desc))
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)